app/src/org/commcare/activities/FormAndDataSyncer.java

```diff
@@ -60,11 +61,11 @@ protected void deliverResult(CommCareHomeActivity receiver, Integer result) {
                     receiver.displayMessage(label);
 
                     if (syncAfterwards) {
-                        syncData(true, userTriggered);
+                        syncData(receiver, true, userTriggered);
```

Comment:
Realizing that this might be a sorta bad bug, where if you perform a sync that submits forms and rotate the screen before the form submission finishes, then the sync task that is launched will have a stale activity.